XRTK-Core/Packages/com.xrtk.core/Runtime/Services/TeleportSystem/MixedRealityTeleportSystem.cs

```diff
@@ -190,45 +196,10 @@ public void RaiseTeleportCanceled(IMixedRealityPointer pointer, IMixedRealityTel
 
             // Pass handler
             HandleEvent(teleportEventData, OnTeleportCanceledHandler);
+
+            isTeleporting = false;
         }
 
         #endregion IMixedRealityTeleportSystem Implementation
-
-        private void ProcessTeleportationRequest(TeleportEventData eventData)
-        {
-            isProcessingTeleportRequest = true;
-
-            targetRotation = Vector3.zero;
-            targetPosition = eventData.Pointer.Result.EndPoint;
-            targetRotation.y = eventData.Pointer.PointerOrientation;
-
-            if (eventData.HotSpot != null)
-            {
-                targetPosition = eventData.HotSpot.Position;
-
-                if (eventData.HotSpot.OverrideTargetOrientation)
-                {
-                    targetRotation.y = eventData.HotSpot.TargetOrientation;
-                }
-            }
-
-            var cameraTransform = MixedRealityToolkit.CameraSystem == null
-                ? CameraCache.Main.transform
-                : MixedRealityToolkit.CameraSystem.MainCameraRig.CameraTransform;
-            var cameraPosition = cameraTransform.position;
-            var cameraParent = cameraTransform.parent;
-            var parentPosition = cameraParent.position;
-            Debug.Assert(cameraParent != null, "The Teleport System requires that the camera be parented under another object.");
-            var height = targetPosition.y + parentPosition.y;
-            targetPosition -= cameraPosition - parentPosition;
-            targetPosition.y = height;
-            cameraParent.position = targetPosition;
-            cameraParent.RotateAround(cameraPosition, Vector3.up, targetRotation.y - cameraTransform.eulerAngles.y);
-
-            isProcessingTeleportRequest = false;
-
-            // Raise complete event using the pointer and hot spot provided.
-            RaiseTeleportComplete(eventData.Pointer, eventData.HotSpot);
```

Comment:
I take this all this was moved to the TeleportHandler implementation?

Reply:
Right